packages/ember/tests/component_context_test.js

```diff
@@ -0,0 +1,145 @@
+import { Controller } from 'ember-runtime';
+import { run } from 'ember-metal';
+import { Component } from 'ember-glimmer';
+import { jQuery } from 'ember-views';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
+
+moduleFor('Application Lifecycle - Component Context', class extends ApplicationTestCase {
+  ['@test Components with a block should have the proper content when a template is provided'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
+
+    this.add('controller:application', Controller.extend({
+      'text': 'outer'
+    }));
+    this.addComponent('my-component', { ComponentClass: Component.extend({
+      text: 'inner'
+    }), template: '{{text}}-{{yield}}' });
+
+    this.visit('/').then(() => {
+      assert.equal(jQuery('#wrapper').text(), 'inner-outer', 'The component is composed correctly');
```

Comment:
`jQuery` here can be replaced with: `this.$('#wrapper')`. The latter will scope the selector (and avoid importing `jQuery`). Also elsewhere through this patch.